### ISSUE TITLE
Support generating from jsdoc classes

### ISSUE THREAD

**Issue by wSedlacek** [mmkal/ts#241]
https://github.com/mmkal/ts/blob/43cf60beb6d0ed7efa349e9c1fe3770d0d9df0e8/packages/eslint-plugin-codegen/src/presets/markdown-from-jsdoc.ts#L25

This is only looking at variables and arrow functions, it would be cool if it could look for classes, abstract classes, interfaces, etc.

**Comment by mmkal** [comment]
Agreed, that would be nice. It doesn't exist now because I personally don't use classes much so didn't have a use case when this was first written. And parsing classes is much harder. Possibly there is a library that could be used to pull out all the comments. A PR would be welcome but it could be a lot of work! In the meantime you may want to look at a more heavyweight tool like https://api-extractor.com/
